src/main/java/it/pagopa/ecommerce/commons/documents/v1/serialization/TransactionEventTypeResolver.java

```diff
@@ -0,0 +1,210 @@
+package it.pagopa.ecommerce.commons.documents.v1.serialization;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.DatabindContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.jsontype.impl.TypeIdResolverBase;
+import it.pagopa.ecommerce.commons.documents.v1.*;
+import it.pagopa.ecommerce.commons.domain.v1.TransactionEventCode;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AssignableTypeFilter;
+
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Jackson type resolver for deserializing {@link TransactionEvent}s
+ */
+public class TransactionEventTypeResolver extends TypeIdResolverBase {
+    private JavaType superType;
+
+    private static final Map<Class<? extends TransactionEvent<?>>, TransactionEventCode> CLASS_TO_EVENT_CODE_MAP = Map
+            .ofEntries(
```

Comment:
This map can be constructed programmatically using getEventClasses method instead of manually populating it, just to be less error prone in mappings, what do you think?

Reply:
This modifications will also simplify below checks removing the one for "all event classes are present inside maps)

Reply:
It can't because event codes don't have a strict naming mapping to classes (e.g. TransactionUserReceiptAddErrorEvent.class <-> TransactionEventCode.TRANSACTION_ADD_USER_RECEIPT_ERROR_EVENT), otherwise it could have been possible (via transforming the event code to PascalCase).

Maybe we could special-handle the few uneven cases and convert to PascalCase the rest if it's not too messy?

Reply:
No I just mean to retrieve all event classes, then instantiate each object reflectively (using the no args constructor) and then retrieve class -> event code mapping using the `getEventCode` method for each instantiated event

Reply:
Oh I like that! It's readable and scalable [with just the right amount of dirty](https://www.youtube.com/watch?v=cjUM3VAk6D4).

Reply:
Solved in https://github.com/pagopa/pagopa-ecommerce-commons/pull/94/commits/49271fbcfaaba634b6804ce9ec9571f5916f2946